lib/internal/readline/interface.js

```diff
@@ -140,6 +140,17 @@ function InterfaceConstructor(input, output, completer, terminal) {
 
   FunctionPrototypeCall(EventEmitter, this);
 
+  {
+    if (Readable === undefined) {
+      Readable = require('stream').Readable;
+    }
+    const onIterator = on(this, 'line');
+    this.once('close', () => onIterator.return());
+    const readable = Readable.from(onIterator).map((line) => line[0]);
+    this[kLineObjectStream] = readable;
```

Comment:
Note the current implementation does _not_ close the `readline.Interface` when the async iterable is closed. This can be easily fixed if we want by doing `readable.once('close', () => this.close())` though I am not sure which behavior is better.

Reply:
I think it should close it. The pattern for async iterators is that the stream is fully consumed after the loop. Maybe add an option?

Reply:
I think it should close it too but was hesitant to change the existing behavior. If you agree it should be closed I'll close it.

Reply:
I think so.